pkg/process/procutil/data_scrubber_windows.go

```diff
@@ -20,3 +25,61 @@ var (
 	// it's for handling parameters like `/p` and `/rp` in windows
 	forbiddenSymbolsRegex = "[^/a-zA-Z0-9_*]"
 )
+
+var winDotExec = []string{".com", ".exe", ".bat", ".cmd", ".vbs", ".vbe", ".js", ".jse", ".wsf", ".wsh", ".psc1", ".ps1"}
+
+// stripArguments identifies windows extension and extracts command. Otherwise, returns the first element of the cmdline before first space.
+func (ds *DataScrubber) stripArguments(cmdline []string) []string {
+	strCmdline := cmdline[0]
+
+	// Case 1: OS has already completed splitting as there is one token per element.
+	if len(cmdline) > 1 {
+		return cmdline
+	}
+
+	// Case 2: One string for cmdline, use extensionParser() to find first token.
+	if len(cmdline) == 1 && !strings.HasPrefix(strCmdline, "\"") {
+		strippedCmdline, foundExtension := extensionParser(strCmdline, winDotExec)
+
+		if foundExtension {
+			return []string{strings.TrimSuffix(strippedCmdline, " ")}
+		}
+
+		// If no extension is found, return first token of cmdline.
+		return []string{strings.TrimSuffix(strings.SplitN(strippedCmdline, " ", 2)[0], " ")}
+	}
+
+	// Case 2b: One string for cmdline and first token wrapped in quotes, use findEmbeddedQuotes() to find content between quotes.
+	strippedCmdline := findEmbeddedQuotes(strCmdline)
+	return []string{strings.TrimSuffix(strippedCmdline, " ")}
+}
+
+// extensionParser returns substring of cmdline up to the first extension (inclusive), and boolean on if extension was found.
+// If no extension is found, returns original cmdline.
+// Example: Input="C:\\Program Files\\Datadog\\agent.vbe check process"  Output="C:\\Program Files\\Datadog\\agent.vbe"
+func extensionParser(cmdline string, winDotExec []string) (string, bool) {
+	for _, c := range winDotExec {
+		// If extension is found before a word break (space or end of line).
+		if i := strings.Index(cmdline, c); i != -1 && (i+len(c) == len(cmdline) || cmdline[i+len(c)] == ' ') {
```

Comment:
I'm not exactly sure, but my intuition says this may be more readable if you use `strings.SplitAfterN` ([doc](https://pkg.go.dev/strings#SplitAfterN))?

Reply:
Just read the documentation for `SplitAfterN` and I see how it could be more readable than using `i+len(c)` to find position after the extension and split based on that!

However, I may be wrong, but I think it will be difficult to use `SplitAfterN` for this case as we need to search for two separators (1. extension such as ".com" for the case when extension is at end of command line, 2. extension followed by a space such as ".com " for the case when there are arguments after the executable path). 
- I think it is possible for windows files to have files named like `"file.executable.exe"` with multiple periods, and in that case using `SplitAfterN` on the separator ".exe" would give incorrect results without additional logic.

An alternative that may be more readable is to use regex to use the word break symbol "\b" to mark end of line or space, but this would require compiling regex for each extension which may be less efficient than using string library functions.

Reply:
mind add a test case with something like your example `"file.executable.exe"`?

Reply:
nvm, I see this test case you have
```
"C:\\Program Files\\Datadog\\agent.exec.process.cmd check process",
```